### ISSUE TITLE
Support for CustomCrafting 2.16.5.0/WU 1.8.0.0

### ISSUE THREAD

**Issue by LcorpOfficial** [Aeltumn/automated_crafting#30]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

This is the current major version of CC and WU used in 1.18. Automated Crafting works so far in 1.18 except for the CC integration

**Describe the solution you'd like**

Autocrafters to work with this version so I will not need to downgrade my CC version and completely redo my recipes (over 30)

**Describe alternatives you've considered**

Suggest a possible less version-specific method to catch these custom recipes from CC perhaps?

**Additional context**

**Comment by Aeltumn** [comment]
CustomCrafting has not yet been updated to 1.18 so I can't test if it works in 1.18, but if it doesn't I'm not going to write specific compatibility. Wolfy keeps rewriting CC and WU every few months and I simply can't keep up with that. Originally I had specific code to be compatible with them, at some point they changed it so all custom recipes were registered to Bukkit which makes them get automatically recognised by AC. I thought finally it would be supported properly.

But if Wolfy no longer registers the recipes with Bukkit and requires me to add custom code again to parse his recipes, I'm not going to bother.

**Comment by LcorpOfficial** [comment]
That is understandable, I'll do some research on how it's handled on Wolfy's side (I thought they were still registered to Bukkit as well) and make an attempt at a fork that works with this compat from here on if needed.

Currently, CC 2.16.5.0 works with my build of Paper 81 for 1.18; short of Mythicmobs items support (that's due to my builds of crucible & MM though, not really CC. Vanilla item recipes still work flawlessly.)

My server uses this as a primary feature for gameplay and with 1.18 on the sights for it when available I might have to learn some actual java haha. great work with the 1.18 update + glow frames and thank you again for your time. This plugin is wonderful and unique with a easy to grasp mechanic. 😁

**Comment by Aeltumn** [comment]
I've just tested CC support and it works fine for me running CC 2.16.6.0-f and WU 2.16.2.0. Both adding a custom recipe for existing items and crafting custom items. (do run /reloadrecipes after modifying anything on CC's end to update the autocrafters)

**Comment by LcorpOfficial** [comment]
Are you testing with Current release of AC (2.6) on 1.18.1 in this instance? My test server is on 1.18.1 Paper but production is still on 1.17.1 (still working out other plugin compat before updating). I've noticed 2.6 of AC would not init on 1.17.1 from my testing. This is fine as I do plan on going to 1.18.1 but I just need to know the details so I can correctly roadmap the updates to my plugins.  

some notes from testing:

AC 2.5 + CC 2.16.7.0 + WU 2.16.2.0 does NOT register custom recipes on 1.17.1

AC 2.6 + CC 2.16.7.0 + WU 2.16.2.0 also does NOT register custom recipes on 1.18.1 either.

ran /reloadrecipes on both instances of testing to be sure.

I can attempt CC 2.16.6.0-f and see if maybe something changed between those versions (I'm not even sure if the non-working mythicmobs items could be affecting it, I'm trying a vanilla item-based CC recipe for invisible item frames using only regular frame + eye of ender, which works in a table.)

not sure if something in my environment is messing with it or what. I'll add a comment shortly after testing CC 2.16.6.0-f. I will have to unfortunately rebuild my CC recipes if downgrading so I'll use a backup for this test and let it update.

**Comment by LcorpOfficial** [comment]
Testing with AC 2.6 + 2.16.6.0-f + WU 2.16.2.0 on 1.18.1 yields strange results. Invisible frame still does not work, but making a new recipe using moss carpet and dirt to make grass block works. removing the recipe for the invis frame and remaking it as a new instance did not work either, the autocrafter does not accept the recipe. the primary difference is materials used and shaped/shapeless recipes.

**Comment by Aeltumn** [comment]
AC 2.6 only works on 1.18.1, it won't work on 1.17. Things worked fine with the latest versions of CC and WU for me. Maybe CC can't register recipes to Bukkit under certain conditions? It worked for me with a custom added recipe for an existing vanilla item and when using their advanced crafting table recipe (the normal crafting recipe for the custom table item, not a recipe on the custom crafting table). But maybe there's something about your invisible item frame item.

**Comment by LcorpOfficial** [comment]
Perhaps; I can say it's simply a named item frame with the invis NBT tag. I can get a `/paper dumpitem` of it later if it would be helpful. I can't test MM items yet with current versions as I don't yet have their latest 1.18.1 dev build just yet (paywalls, lol) But I would think they SHOULD work the same as an Invis frame vs using purely vanilla stuff. just thinking out loud. those items are a bit more complicated with lore text, NBT tags, and the like. 

I do know CC has a bit where it can register recipes to the vanilla crafting book but I think only if the ingredients are vanilla stuff, no custom ingredients allowed. resulting items do not seem to matter. Not normally important to mention but I wonder if perhaps there is something special about the workflow on CC's side regarding actually registering the recipes. I could be shooting entirely in the dark here though, so grain of salt.

**Comment by LcorpOfficial** [comment]
Yeah, Wolfy released an update to WU and CC for "compatibility fixes", mostly for 1.16 servers, but it seems that the invis frame, is now the only recipe not working. only certain mythicmobs items, and vanilla item-based recipes are working with current release of MM, WU, CC, and AC as of the writing of this. Kind of strange. I think you are right, it does seem conditional. I have noticed that recipes that require custom items ingredients (ones incompatible with the vanilla recipe book) do not register in AC

**Comment by Aeltumn** [comment]
Okay so I've figured out that the only way to support CC is to re-add custom compatibility logic in the future. So I'll re-open this issue as a reminder to look into it. CC only puts placeholder recipes into the registry which are stripped of all custom NBT. This means that AC can only properly auto-craft recipes where none of the ingredients have special data. The only way for me to support this special data is to add specific compatibility, but I don't know if/when I'll do that.

**Comment by LcorpOfficial** [comment]
that makes sense, probably changed to that when they added support for vanilla crafting table listing support perhaps. Please update here with a comment when you reach a verdict on the compact decision, and thanks so much for the time you've invested in this plugin. it's a game-changer in my server. If you don't end up doing the compat I might end up having to learn enough Java to make a fork, if that would be acceptable.

**Comment by LcorpOfficial** [comment]
Just a note: it seems Wolfe updated CC to change how recipes are serialized and stored, and perhaps could be easier to adapt either natively or I could make an adapter worst case scenario
